qa/tasks/cephfs/test_volumes.py

```diff
@@ -127,6 +127,60 @@ def test_subvolume_create_and_rm_in_group(self):
         # remove group
         self._fs_cmd("subvolumegroup", "rm", self.volname, group)
 
+    def test_subvolume_group_create_with_desired_data_pool_layout(self):
+        group1 = self._generate_random_group_name()
+        group2 = self._generate_random_group_name()
+
+        # create group
+        self._fs_cmd("subvolumegroup", "create", self.volname, group1)
+        group1_path = os.path.join('volumes', group1)
+
+        default_pool = self.mount_a.getfattr(group1_path, "ceph.dir.layout.pool")
+        new_pool = "new_pool"
+        self.assertNotEqual(default_pool, new_pool)
+
+        # add data pool
+        self.fs.add_data_pool(new_pool)
+
+        # create group specifying the new data pool as its pool layout
+        self._fs_cmd("subvolumegroup", "create", self.volname, group2,
+                     "--pool_layout", new_pool)
+        group2_path = os.path.join('volumes', group2)
+
+        desired_pool = self.mount_a.getfattr(group2_path, "ceph.dir.layout.pool")
+        self.assertEqual(desired_pool, new_pool)
+
+    def test_subvolume_create_with_desired_data_pool_layout_in_group(self):
+        subvol1 = self._generate_random_subvolume_name()
+        subvol2 = self._generate_random_subvolume_name()
+        group = self._generate_random_group_name()
+
+        # create group. this also helps set default pool layout for subvolumes
+        # created within the group.
+        self._fs_cmd("subvolumegroup", "create", self.volname, group)
+
+        # create subvolume in group.
+        self._fs_cmd("subvolume", "create", self.volname, subvol1, "--group_name", group)
+        subvol1_path = self._fs_cmd("subvolume", "getpath", self.volname, subvol1, group)
+
+        # remove leading '/' and trailing whitespaces from subvolume path
+        default_pool = self.mount_a.getfattr(subvol1_path[1:].rstrip(),
```

Comment:
nit: do the `[1:].rstrip()` once and use wherever needed. +1 if you can carve it out into a routine since it seems to be needed in few places.

Reply:
[1:].rstrip() is used used only once in this method for extracting the subvol1 path needed by the mount object. 

Are you OK with this routine?

```
+ def  _get_subvolume_mount_path(self, vol_name, subvol_name, group_name=None):
+        args = ["subvolume", "getpath", vol_name, subvol_name]
+        if group_name:
+            args.append(group_name)
+        args = tuple(args)
+        path = self._fs_cmd(*args)
+        return path[1:].rstrip()
```

Reply:
yep, `_get_subvolume_path()` should be good enough.

Reply:
Done